docs/architecture/grpc-for-wcf-developers/error-handling.md

```diff
@@ -7,5 +7,65 @@ ms.date: 09/02/2019
 
 # Error handling
 
+WCF uses `FaultException<T>` and `FaultContract` to provide detailed error information, including supporting the SOAP Fault standard.
+
+Unfortunately, the current version of gRPC lacks the sophistication found with WCF and only has limited built-in error handling based on simple status codes and metadata. The following table is a quick guide to the most commonly used status codes:
+
+| Status Code | Problem |
+| ----------- | ------- |
+| `GRPC_STATUS_UNIMPLEMENTED` | Method hasn’t been written. |
+| `GRPC_STATUS_UNAVAILABLE` | Problem with the whole service. |
+| `GRPC_STATUS_UNKNOWN` | Invalid response. |
+| `GRPC_STATUS_INTERNAL` | Problem with encoding/decoding. |
+| `GRPC_STATUS_UNAUTHENTICATED` | Authentication failed. |
+| `GRPC_STATUS_PERMISSION_DENIED` | Authorization failed. |
+| `GRPC_STATUS_CANCELLED` | Call was canceled, usually by the caller. |
+
+## Raising errors in ASP.NET Core gRPC
+
+An ASP.NET Core gRPC service can send an error response by throwing an `RpcException`, which can be caught by the client as if it were in the same process. The `RpcException` must include a status code and description, and can optionally include metadata and a longer exception message. The metadata can be used to send supporting data, similar to how `FaultContract` objects could carry additional data for WCF errors.
+
+```csharp
+public async Task<GetPortfolioResponse> GetPortfolio(GetPortfolioRequest request, ServerCallContext context)
+{
+    var user = context.GetHttpContext().User;
+    if (!ValidateUser(user))
+    {
+        var metadata = new Metadata
+        {
+            { "User", user.Identity.Name }
+        };
+        throw new RpcException(new Status(StatusCode.PermissionDenied, "Permission denied"), metadata);
+    }
+}
+```
+
+## Catching errors in gRPC clients
+
+Just like WCF clients can catch <xref:System.ServiceModel.FaultException%601> errors, a gRPC client can catch an `RpcException` to handle errors. Since `RpcException` isn't a generic type, you can't catch different error types in different blocks, but you can use C#'s *exception filters* feature to declare separate `catch` blocks for different status codes, as shown in the following example:
```

Comment:
Does the %601 form part of the xref? Just want to make sure this isn't accidental junk.

Reply:
Yes. We need to encode the backtick in the xref links.